src/components/Billing/Invoice/EditInvoiceTable.tsx

```diff
@@ -257,54 +425,189 @@ export function EditInvoiceTable({
-                        )}
+                        name={`items.${index}.discountComponentKey`}
+                        render={({ field }) => {
+                          const chargeItem = chargeItems[index];
+                          const availableDiscounts =
+                            getAvailableDiscounts(chargeItem);
+
+                          // Get the current discount component to show its name
+                          const currentDiscount = availableDiscounts.find(
+                            (c) => getDiscountComponentKey(c) === field.value,
+                          );
+
+                          return (
+                            <FormItem className="flex-1">
+                              <FormControl>
+                                <Select
+                                  value={field.value || "none"}
+                                  onValueChange={(value) => {
+                                    if (value === "none") {
+                                      form.setValue(
+                                        `items.${index}.discountComponentKey`,
+                                        undefined,
+                                      );
+                                      form.setValue(
+                                        `items.${index}.discountCode`,
+                                        undefined,
+                                      );
+                                      form.setValue(
+                                        `items.${index}.discountConditions`,
+                                        [],
+                                      );
+                                      form.setValue(
+                                        `items.${index}.discountValue`,
+                                        "0",
+                                      );
+                                    } else {
+                                      handleDiscountComponentChange(
+                                        index,
+                                        value,
+                                      );
+                                    }
+                                  }}
+                                >
+                                  <SelectTrigger>
+                                    <SelectValue>
+                                      {currentDiscount ? (
+                                        <>
+                                          {currentDiscount.code?.display} @{" "}
+                                          {currentDiscount.factor ??
+                                            currentDiscount.amount ??
+                                            0}
+                                          {currentDiscount.factor != null
+                                            ? "%"
+                                            : "₹"}
+                                        </>
+                                      ) : (
```

Comment:
Pure mess, we have a component to render money!
Also how do we know its INR always?